tgf_auto_update.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+// RunUpdate runs the update on the current tgf executable
+func RunUpdate() bool {
+	executablePath, err := os.Executable()
```

Comment:
Since this is a OS-specific operation. You can move all of this code to `auto_update_unix.go` and implement the same functions in `auto_update_windows.go` but they should be empty and log (debug) `Auto update not implemented on windows` or something